tests/tape/test_qnode.py

```diff
@@ -573,6 +573,39 @@ def func():
         # No differentiation required. No error raised.
         grad()
 
+    @pytest.mark.parametrize("par", ["QNode", "decorator"])
+    def test_unrecognized_keyword_arguments_validation(self, par):

```

Comment:
Thanks for parametrizing this, although, looking at it, I think it'd be better just to split them into two different tests instead. Basically just splitting the 
```python
            if par == "QNode":
                QNode(dummyfunc, dev, test_method_one=1, test_method_two=2)
            elif par == "decorator":

                @qml.qnode(dev, test_method_one=1, test_method_two=2)
                def circ():
                    return qml.expval(qml.PauliZ(0))
```
into two different tests instead of the if-statement (even though there's some code duplication, it tests two different functions, and should probably be tested as separate tests).

Reply:
@thisac yeah, that sounds good. I was debating that but I agree two tests sounds good.